app/models/comment.rb

```diff
@@ -26,17 +28,19 @@ class Comment < ApplicationRecord
   before_create :adjust_comment_parent_based_on_depth
   after_create :after_create_checks
   after_create :notify_slack_channel_about_warned_users
-  after_update :update_descendant_notifications, if: :deleted
   after_update :remove_notifications, if: :deleted
+  after_update :update_descendant_notifications, if: :deleted
   before_destroy :before_destroy_actions
   after_destroy :after_destroy_actions
+
   after_save :synchronous_bust
   after_save :bust_cache
-  validate :permissions, if: :commentable
+
+  validate :published_article, if: :commentable
   validates :body_markdown, presence: true, length: { in: BODY_MARKDOWN_SIZE_RANGE }
   validates :body_markdown, uniqueness: { scope: %i[user_id ancestry commentable_id commentable_type] }
-  validates :commentable_id, presence: true
-  validates :commentable_type, inclusion: { in: COMMENTABLE_TYPES }
+  validates :commentable_id, presence: true, if: :commentable_type
+  validates :commentable_type, inclusion: { in: COMMENTABLE_TYPES }, if: :commentable_id
```

Comment:
We declare `commentable` as `optional: true` on line 16 but then, until now, we requested `commentable_id` to be present.

We should check its presence only if `commentable_type` is present as well and we should check the valid type only if `commentable_id` is there